catalogue_of_life/Loader.java

```diff
@@ -0,0 +1,242 @@
+
+        Record record;
+        Map<String, String> recordValues = new HashMap<>();
+        while ((record = distributionsParser.parseNextRecord()) != null) {
+            String gazetteer = record.getString("col:gazetteer");
+            if (gazetteer.equals("text") || gazetteer.equals("iso")) {
+                record.fillFieldMap(recordValues);
+                describedRegionsWriter.writeRow(recordValues);
+            } else if (gazetteer.equals("mrgid") && isValidMarineRegionID(record.getString("col:areaID"))) {
+                record.fillFieldMap(recordValues);
+                marineRegionsWriter.writeRow(recordValues);
+            }
+        }
+
+        marineRegionsWriter.close();
+        describedRegionsWriter.close();
+    }
+
+    private static TsvParser beginParsingFile(Path tsvFile) throws IOException {
+        TsvParserSettings settings = new TsvParserSettings();
+        settings.setHeaderExtractionEnabled(true);
+        settings.setMaxCharsPerColumn(1_000_000);
+
+        TsvParser parser = new TsvParser(settings);
+        InputStream inputStream = Files.newInputStream(tsvFile);
+        parser.beginParsing(inputStream);
+
+        return parser;
+    }
+
+    private static TsvWriter createTsvWriter(Path tsvFile) throws IOException {
+        TsvWriterSettings writerSettings = new TsvWriterSettings();
+        writerSettings.setHeaderWritingEnabled(true);
+        writerSettings.setMaxCharsPerColumn(1_000_000);
+
+        return new TsvWriter(Files.newOutputStream(tsvFile), writerSettings);
+    }
+
+    private static boolean isValidMarineRegionID(String areaID) {
+        return isNumeric(areaID);
+    }
+
+    private static boolean isNumeric(String s) {
+        try {
+            Long.parseLong(s);
+        } catch (NumberFormatException e) {
+            return false;
+        }
+        return true;
+    }
+
+    public static void loadData(Path dataDirectory, Options options) throws IOException {
+        extractSchema(dataDirectory);
+
+
+        LoadOptions loadOptions = new LoadOptions();
+        loadOptions.dataConfigFilePath = createTypeDBLoaderConfig(dataDirectory).toString();
+        loadOptions.databaseName = "catalogue-of-life";
+        loadOptions.typedbURI = options.typedbURI();
+        loadOptions.cleanMigration = true;
+        loadOptions.loadSchema = false;
+
+        TypeDBLoader loader = new TypeDBLoader(loadOptions);
+        loader.load();
+    }
+
+    public static void extractSchema(Path dataDirectory) throws IOException {
+        InputStream schema = Loader.class.getResourceAsStream("/catalogue_of_life/schema.tql");
+        if (schema == null) throw new FileNotFoundException("schema.tql not found within the jar!");
+        Files.copy(schema, dataDirectory.resolve("schema.tql"), StandardCopyOption.REPLACE_EXISTING);
+    }
+
+    public static Path createTypeDBLoaderConfig(Path dataDirectory) throws IOException {
```

Comment:
maybe `populateLoaderConfigTemplate` is better?
If you call it `create` it sounds like the return type should be a `Config`. Additionally, really what you're doing is filling in a template as the act of creation so including the word `template` is more descriptive than `create` and tells the reader what this function is doing more succinctly